packages/core/src/lib/position.mts

```diff
@@ -0,0 +1,49 @@
+type Change<T> = {
+  previous: T;
+  current: T;
+};
+
+type PageInfo = {
+  active: Change<number>;
+  total: number;
+  pageSize: number;
+};
+
+/**
+ * Given information about a page, decides the next position at which the active
+ * item should be rendered in the page.
+ */
+type PositionReducer = (info: PageInfo) => (pointer: number) => number;
+
+/**
+ * Creates the next position for the active item considering a finite list of
+ * items to be rendered on a page.
+ */
+export const finite: PositionReducer =
+  ({ pageSize, total, active }) =>
+  () => {
```

Comment:
More of a `nit`, but I don't see the value of most of those curried functions (with their current usage inside the hook.)

It feels like unnecessary complexity VS passing 2 arguments to the function.

Reply:
I agree, I was more thinking about `setPointer(current => next)` usage at some point, when being able to do
`setPointer(infinite(args))` might be useful. I'll bring this forward after a few more iterations and you can see if you like it. Otherwise we'll undo the curried version and make it simpler.